### PR TITLE
Move XDG user-dir helpers out of state.rs (Wave 3, #38)

### LINKED ISSUE #38: Move XDG user-dir helpers out of `state.rs`
- **Severity:** low · **Effort:** M · **Labels:** `refactor`
- **Source:** B-F10, D-F10, E-F1

40% of `state.rs` is one-shot XDG `user-dirs.dirs` parsing (`map_xdg_user_dirs` + `parse_user_dir_line`). One-shot config load is not state. First check `nwg_common::config::paths` for an upstream-able home; otherwise move to `src/xdg_dirs.rs`. Also: `parse_user_dir_line` discards parse-failure context (no log even at debug) — fix while moving.

**Acceptance:** `state.rs` is data-only; helpers either upstreamed or in their own module; `parse_user_dir_line` logs malformed lines at `debug`; new pure-function tests against the parser (combines E-F1 here so we get coverage in the same PR).

---

Part of #28. Full review context: [`docs/code-review-2026-05.md`](https://github.com/jasonherald/nwg-drawer/blob/main/docs/code-review-2026-05.md).

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Closes #38. **Final Wave 3 architectural cleanup.** With this merged, Wave 3 is complete (5/5).

`state.rs` had ~40% of its bulk in one-shot XDG `~/.config/user-dirs.dirs` parsing — `map_xdg_user_dirs` + `parse_user_dir_line`. One-shot config load isn't state. Verified `nwg-common` has no upstream home for this (no XDG user-dirs handling anywhere in `nwg_common::config` or `nwg_common::desktop`), so going local: new `src/xdg_dirs.rs`.

## Layout after

| File | Lines | Purpose |
|---|---|---|
| `src/state.rs` | 112 → **98** | `DrawerState` / `AppRegistry` definitions only — data shape, no logic |
| `src/xdg_dirs.rs` | new, **157** | Parser + 7 unit tests |

## Behavioral cleanup while moving

Per the issue body's "fix while moving" instruction:

- **`parse_user_dir_line` now logs at `debug` on malformed input.** Two log sites: missing `=`, and unparseable value after a recognized XDG key. Comments and blanks are silently skipped — those are normal in the file format.
- **Empty value returns `None`.** `XDG_DIR=""` or `XDG_DIR=` previously produced a zero-length `PathBuf`, which would have inserted an empty path into the registry. Now those lines are skipped.
- **XDG-key dispatch collapses** from five `else if line.starts_with(...)` arms to one `match` on the parsed key.

Same behavior on the success path; only the error/edge cases were tightened.

## Tests

7 new pure-function tests cover:

- Quoted value with `$HOME` expansion
- Unquoted value
- Line without `=` returns `None`
- Empty value returns `None` (both `=""` and `=`)
- `$HOME` substitution works anywhere in the path (pinned, since this is a quirk of the implementation)
- Path with no `$HOME` reference passes through unchanged
- Whitespace around the value is trimmed

This combines the E-F1 acceptance ("new pure-function tests against the parser") with the move.

## Acceptance from #38

- [x] `state.rs` is data-only
- [x] Helpers in their own module (no upstream home in nwg-common — checked)
- [x] `parse_user_dir_line` logs malformed lines at `debug`
- [x] New pure-function tests against the parser

## Test plan

- [x] `cargo build` clean
- [x] `make lint` clean (fmt + clippy + test + deny + audit)
- [x] All 89 tests pass — 82 existing + 7 new

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Support for reading and applying XDG user directory overrides from user XDG config (e.g., custom Documents/Downloads/Pictures locations).

* **Refactor**
  * Directory path resolution moved into a dedicated component for cleaner organization and reuse.

### CODE REVIEW

File: src/xdg_dirs.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    result.insert("home".into(), PathBuf::from(&home));
+    result.insert("documents".into(), PathBuf::from(&home).join("Documents"));
+    result.insert("downloads".into(), PathBuf::from(&home).join("Downloads"));
+    result.insert("music".into(), PathBuf::from(&home).join("Music"));
+    result.insert("pictures".into(), PathBuf::from(&home).join("Pictures"));
+    result.insert("videos".into(), PathBuf::from(&home).join("Videos"));
+
+    let config_home =
+        std::env::var("XDG_CONFIG_HOME").unwrap_or_else(|_| format!("{}/.config", home));
+    let user_dirs_file = PathBuf::from(&config_home).join("user-dirs.dirs");
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply src/xdg_dirs.rs]: Fixed in 633515a — the function now returns an empty map with `log::warn!` if HOME is missing/empty. The consumer at file_search.rs:25 naturally skips iteration on an empty map, so no file search runs against bogus relative paths. Applied your suggested guard shape.

File: src/xdg_dirs.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                None => {
+                    log::debug!(
+                        "Skipping unparseable user-dirs.dirs line for {}: {}",
+                        key,
+                        line
+                    );
+                }
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply src/xdg_dirs.rs]: Fixed in 633515a — `if let Ok` replaced with explicit match. NotFound stays silent (common "never ran xdg-user-dirs-update" case); other I/O errors now `log::debug!` with the file path and error. Took the opportunity to extract the override loop into `apply_overrides(&mut result, content, home)` since the surrounding logic was getting busier.

File: src/xdg_dirs.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                "XDG_DOCUMENTS_DIR" => "documents",
+                "XDG_DOWNLOAD_DIR" => "downloads",
+                "XDG_MUSIC_DIR" => "music",
+                "XDG_PICTURES_DIR" => "pictures",
+                "XDG_VIDEOS_DIR" => "videos",
+                _ => continue,
+            };
+            match parse_user_dir_line(line, &home) {
+                Some(val) => {
+                    result.insert(bucket.into(), val);
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply src/xdg_dirs.rs]: Fixed in 633515a. Introduced `pub enum XdgDirBucket { Home, Documents, Downloads, Music, Pictures, Videos }` with derived Hash + Eq + Copy. The override match now maps `XDG_DOCUMENTS_DIR` → `XdgDirBucket::Documents` directly. `HashMap<String, PathBuf>` → `HashMap<XdgDirBucket, PathBuf>` propagates to `DrawerState::user_dirs` and the single consumer at `file_search.rs:26` (`dir_name == "home"` → `*bucket == XdgDirBucket::Home`, typo-proof at compile time). Bigger scope than the issue body originally asked for, but stays in the same conceptual neighborhood and matches CLAUDE.md "enums over strings."
